mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm

```diff
@@ -3943,6 +3943,37 @@ var XPIInstall = {
     return addon;
   },
 
+  async getCliqzAddonFromLocation(loc) {
+    const addons = addonMap(
+      await XPIDatabase.getAddonsInLocation(loc)
+    );
+    return addons.get("cliqz@cliqz.com");
+  },
+
+  async updateCliqzToLatest() {
+    let cliqzFromProfile = await this.getCliqzAddonFromLocation(KEY_APP_PROFILE) || {};
+    let cliqzFromSystem = await this.getCliqzAddonFromLocation(KEY_APP_SYSTEM_ADDONS) || {};
+    let cliqzFromFeatures = await this.getCliqzAddonFromLocation(KEY_APP_SYSTEM_DEFAULTS) || {};
+    let latestCliqz;
+
+    try {
+      if (cliqzFromProfile.version > cliqzFromFeatures.version) {
+        latestCliqz = cliqzFromProfile;
+      } else {
+        latestCliqz = cliqzFromFeatures;
+      }
+
+      if (latestCliqz.version > cliqzFromSystem.version) {
+        let systemAddonLocation = XPIStates.getLocation(KEY_APP_SYSTEM_ADDONS);
+        await systemAddonLocation.installer.installAddonSet(
+          [latestCliqz]
+        );
+      }
+    } catch(e) {
+      // :(
```

Comment:
@tsl143 could you please log the situation in case we fall down to the catch clause?
I think it might be easier for us to debug it.